.github/workflows/ci.yaml

```diff
@@ -0,0 +1,91 @@
+name: "CI"
+
+# Triggered for commits/PRs on master branch
+# + new tags
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master 
+  create:
+    tags:
+      - v*
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  # Cancel previous actions
+  cleanup:
+    name: "Cleanup"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Get workflow id"
+        run: |
+          WORKFLOW_ID=$(curl "Authorization: token ${{ github.token }}" https://api.github.com/repos/${{ github.repository }}/actions/workflows/ci.yaml | jq '.id')
+          echo "Workflow id: ${WORKFLOW_ID}"
+          echo "WORKFLOW_ID=${WORKFLOW_ID}" >> $GITHUB_ENV
+
+      - uses: styfle/cancel-workflow-action@0.3.1
+        with:
+          access_token: ${{ github.token }}
+          workflow_id: ${{ env.WORKFLOW_ID }}
+
+  # Main build
+  build:
+    name: "Build"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+    steps:
+      # Checkout
+      - name: Checkout
+        uses: actions/checkout@main
+        with:
+          submodules: recursive
+
+      # Set up cache for dependencies
+      - name: Set up cache
+        id: cache-dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            $HOME/.ghc
+            $HOME/.cabal
+            $HOME/.stack
+          key: ${{ runner.os }}-haskell
+
+      # Setup Haskell
+      - name: Set up environment
+        uses: actions/setup-haskell@v1.1
+        with:
+          ghc-version: 'latest'
+          enable-stack: true
+          stack-version: 'latest'
+
+      # Verify everything's right
+      - name: Verify environment
+        run: |
+          stack --numeric-version
+          ghc --version
+          
+      # Build dependencies
+      - name: Build dependencies 
+        run: |
+          cd waspc
+          stack --no-terminal test --only-dependencies
```

Comment:
You removed --install-ghc, but I would put it back. It does nothing if ghc is already installed, so it does not hurt, and it does make this command a little bit more robust to any changes in the previous actions.